src/viam/resource/manager.py

```diff
@@ -91,21 +91,23 @@ def get_resource(self, of_type: Type[ResourceType], name: ResourceName) -> Resou
                 return self.get_resource(of_type, self._short_to_long_name[name.name][0])
             raise ResourceNotFoundError(name.subtype, name.name)
 
-    def remove_resource(self, name: ResourceName):
+    async def remove_resource(self, name: ResourceName):
         """Remove the resource with the specified ```ResourceName```.
 
         Args:
             name (viam.proto.common.ResourceName): The ResourceName of the resource
         """
         with self._lock:
-            del self.resources[name]
+            try:
+                resource = self._resource_by_name_only(name.name)
```

Comment:
I don't think this is the best way to index into a resource - what if we have an arm called `arm1` and a base called `arm1`?